server/sonar-server/src/main/java/org/sonar/server/computation/queue/InternalCeQueueImpl.java

```diff
@@ -107,6 +112,31 @@ private static void updateTaskResult(CeActivityDto activityDto, @Nullable CeTask
     }
   }
 
+  private void updateError(CeActivityDto activityDto, @Nullable Throwable error) {
+    if (error == null) {
+      return;
+    }
+
+    activityDto.setErrorMessage(error.getMessage());
+    String stacktrace = getStackTraceForPersistence(error);
+    if (stacktrace != null) {
+      activityDto.setErrorStacktrace(stacktrace);
+    }
+  }
+
+  @CheckForNull
+  private static String getStackTraceForPersistence(Throwable error) {
+    try (ByteArrayOutputStream out = new ByteArrayOutputStream();
+      LineReturnControllerPrintStream printStream = new LineReturnControllerPrintStream(out);) {
+      error.printStackTrace(printStream);
+      printStream.flush();
+      return out.toString();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this use of "toString" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1943)